crates/bifrost/src/providers/replicated_loglet/tasks/get_trim_point.rs

```diff
@@ -0,0 +1,194 @@
+                header: LogServerRequestHeader::new(
+                    self.my_params.loglet_id,
+                    self.known_global_tail.latest_offset(),
+                ),
+            };
+
+            inflight_requests.spawn({
+                let networking = networking.clone();
+                let trim_rpc_router = self.logservers_rpc.get_loglet_info.clone();
+                let known_global_tail = self.known_global_tail.clone();
+                let tc = task_center();
+
+                async move {
+                    let task = RunOnSingleNode::new(
+                        node_id,
+                        request,
+                        &trim_rpc_router,
+                        &known_global_tail,
+                        Configuration::pinned()
+                            .bifrost
+                            .replicated_loglet
+                            .log_server_retry_policy
+                            .clone(),
+                    );
+                    (
+                        node_id,
+                        tc.run_in_scope(
+                            "find-trimpoint-on-node",
+                            None,
+                            task.run(on_info_response, &networking),
+                        )
+                        .await,
+                    )
+                }
+            });
+        }
+
+        let mut nodeset_checker = NodeSetChecker::<'_, Option<LogletOffset>>::new(
+            &effective_nodeset,
+            &networking.metadata().nodes_config_ref(),
+            &self.my_params.replication,
+        );
+
+        let predicate = |o: &Option<LogletOffset>| o.is_some();
+        // Waiting for trim responses
+        while let Some(res) = inflight_requests.join_next().await {
+            let Ok((node_id, res)) = res else {
+                // task panicked or runtime is shutting down.
+                continue;
+            };
+            let Ok(res) = res else {
+                // GetLogletInfo task failed/aborted on this node. The inner task will log the error in this case.
+                continue;
+            };
+
+            // either f-majority or write-quorum is enough
+            nodeset_checker.set_attribute(node_id, Some(res));
+            if nodeset_checker.check_write_quorum(predicate)
+                || nodeset_checker.check_fmajority(predicate).passed()
+            {
+                break;
+            }
+        }
+
+        let results_from_nodes: Vec<_> = nodeset_checker.filter(predicate).collect();
+        if results_from_nodes.is_empty() {
+            // We didn't get _any_ responses!
+            return Err(OperationError::retryable(GetTrimPointError));
+        }
+
+        let max_trim_point = nodeset_checker
+            .filter(predicate)
+            .map(|(_, o)| o.unwrap()) // we filter Some already
+            .max()
+            .expect("at least one node returned trim point");
+        // Not enough nodes have successful responses
```

Comment:
Is this comment still correct?

Reply:
Not any more, my original plan was to fail if we couldn't hear enough results. After thinking about it, I decided that it's not necessary.